engine/src/main/java/org/destinationsol/game/particle/ParticleEmitter.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.destinationsol.game.particle;
+
+import com.badlogic.gdx.math.Vector2;
+import org.destinationsol.common.NotNull;
+import org.destinationsol.game.SolGame;
+import org.destinationsol.game.drawables.Drawable;
+import org.destinationsol.game.drawables.DrawableLevel;
+import org.destinationsol.game.item.Engine;
+import org.destinationsol.game.ship.SolShip;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ParticleEmitter {
+
+    private final @NotNull Vector2 position;
+    private final @NotNull String particleName, trigger;
+    private final List<Drawable> myDrawables;
+    private final ParticleSrc myParticle;
+
+    public ParticleEmitter(SolGame game, Engine engine, ParticleEmitter particleEmitter, SolShip ship) {
+        myDrawables = new ArrayList<>();
+        EffectConfig effectConfig = engine.getEffectConfig();
+        Vector2 particlePos = particleEmitter.getPosition();
+        Vector2 shipPos = ship.getPosition();
+        Vector2 shipSpd = ship.getSpd();
```

Comment:
`shipSpeed`.
I realize you used the `getSpd()` as inspiration, but I prefer clear names. We'll get around to renaming the correponding method in ship eventually. The least we can do is make new code clear.